From eaae034eb70f3b1bb1bbb502d1aa5d508e3891ef Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 18 May 2026 07:42:32 +0000
Subject: [PATCH] pwm: Fix double shift bug
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

jira VULN-35725
cve CVE-2023-52756
commit-author Dan Carpenter <dan.carpenter@linaro.org>
commit d27abbfd4888d79dd24baf50e774631046ac4732

These enums are passed to set/test_bit().  The set/test_bit() functions
take a bit number instead of a shifted value.  Passing a shifted value
is a double shift bug like doing BIT(BIT(1)).  The double shift bug
doesn't cause a problem here because we are only checking 0 and 1 but
if the value was 5 or above then it can lead to a buffer overflow.

	Signed-off-by: Dan Carpenter <dan.carpenter@linaro.org>
	Reviewed-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
	Reviewed-by: Sam Protsenko <semen.protsenko@linaro.org>
	Signed-off-by: Thierry Reding <thierry.reding@gmail.com>
(cherry picked from commit d27abbfd4888d79dd24baf50e774631046ac4732)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 include/linux/pwm.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/pwm.h b/include/linux/pwm.h
index a0b7e43049d5..1afaf5401ec6 100644
--- a/include/linux/pwm.h
+++ b/include/linux/pwm.h
@@ -44,8 +44,8 @@ struct pwm_args {
 };
 
 enum {
-	PWMF_REQUESTED = 1 << 0,
-	PWMF_EXPORTED = 1 << 1,
+	PWMF_REQUESTED = 0,
+	PWMF_EXPORTED = 1,
 };
 
 /*
-- 
2.47.3

